Source/ACE/Factories/SpellFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿using ACE.Entity;
+using ACE.Entity.Enum;
+using ACE.Network.Enum;
+
+namespace ACE.Factories
+{
+    /// <summary>
+    /// factory class for creating objects related to administration or content-recreation
+    /// </summary>
+    public class SpellFactory
+    {
+        /// <summary>
+        /// creates a spell with velocity directly in front of the player's position provided
+        /// </summary>
+        public static ImmutableWorldObject CreateCastedSpell(ushort weenieClassId, Position newPosition, ObjectGuid objectid, AceVector3 velocity)
```

Comment:
Spells are very much mutable.  They have motion and do things, so use MutableObject.